src/encoding/encoding.rs

```diff
@@ -14,15 +14,15 @@
 //
 // Also licensed under MIT license, at your choice.
 
-use super::resource::Patch;
 use super::{
-    DrawColor, DrawLinearGradient, DrawRadialGradient, DrawTag, PathEncoder, PathTag, Transform,
+    resolve::Patch, DrawColor, DrawLinearGradient, DrawRadialGradient, DrawTag, Glyph, GlyphRun,
+    PathEncoder, PathTag, Transform,
 };
 
-use peniko::{kurbo::Shape, BlendMode, BrushRef, Color, ColorStop, Extend, GradientKind};
+use peniko::{kurbo::Shape, BlendMode, BrushRef, ColorStop, Extend, GradientKind};
 
 /// Encoded data streams for a scene.
-#[derive(Default)]
+#[derive(Clone, Default)]
```

Comment:
I've got my head pretty much around this now, but it's reminding me that we should have a writeup of how these pieces fit together. There's a higher level theory about how glyph runs contain splice points referencing the main encodings, and also lower level details that are significant: for example, conceptually each glyph run contains a sequence of glyphs, but in practice the storage for those is flattened into a single vec. Perhaps this is the proper scope for the big Vello writeup, perhaps we should do something before then.

I'm also aware that the details will change when we get images done.

Reply:
This makes sense. I have a bit of a document queue right now, but I'll try to at least put together some loose notes about this to capture what's happening.